loadbalancer/etcd/client.go

```diff
@@ -0,0 +1,83 @@
+package etcd
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/coreos/go-etcd/etcd"
+)
+
+// EtcdClient is a wrapper arround the etcd client
+type EtcdClient interface {
+	// GetEntries wil query the given prefix in etcd and returns a set of entries.
+	GetEntries(prefix string) ([]string, error)
+	// WatchPrefix starts watching every change for given prefix in etcd. When a
+	// change is detected it will populate the responseChan when an *etcd.Response.
+	WatchPrefix(prefix string, responseChan chan *etcd.Response)
+}
+
+type etcdClient struct {
+	*etcd.Client
+}
+
+// NewClient returns an *etcd.Client with a connection to named machines.
+// It will return an error if a connection to the cluster cannot be made.
+func NewClient(machines []string, cert, key, caCert string) (*etcdClient, error) {
```

Comment:
It may be worth documenting that machines should be full URLs with schemas, e.g. "http://localhost:4001" will work, but "localhost:4001" will not.